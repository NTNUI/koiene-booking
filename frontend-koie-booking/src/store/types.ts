// rootState = generic type, defines the root state type
import AdminBooking, { AdminBookingDictionary } from '@/types/admin/AdminBooking';
import { AdminReportKoieDictionary } from '@/types/admin/AdminReports';

export interface RootState {
  version: string;
  koie: KoieState;
  booking: BookingState;
  auth: AuthState;
  avalanche: AvalancheState;
  adminBookings: AdminBookingsState;
}

export interface KoieState {
  currentLocale: string;
  isLoading: boolean;
  koieData: Koie;
  error: boolean;
  allKoier: [];
}

export interface Koie {
  location: {
    latitude: string;
    longitude: string;
    area: string;
    difficultyInfo_eng: string;
    difficultyInfo_nor: string;
    altitude: number;
    terrain_eng: string;
    terrain_nor: string;
    map_pdf: string;
    kartblad: string;
  };
  description: {
    yr_link_eng: string;
    yr_link_nor: string;
    description_eng: string;
    description_nor: string;
    directions_eng: string;
    directions_nor: string;
    parking_eng: string;
    parking_nor: string;
  };
  beds_available_in_booking_window: Array<BedsAvailable>;
  name: string;
  number_of_beds: number;
  booking_window: number;
  price_member: number;
  price_not_member: number;
}

export interface BedsAvailable {
  date: string;
  bedsAvailable: number;
}

export interface BookingState {
  isLoading: boolean;
  dateFrom: string;
  dateTo: string;
  beds: number;
  step: number;
  guests: Array<Guest>;
  numberOfMembers: number;
  numberOfNonMembers: number;
  edited: boolean;
  validForm: boolean;
  unavailableDates: [];
  availableBeds: number;
  noDatesAvailable: boolean;
  error: boolean;
  bookingData: BookingInfo;
  paymentConfirmed: boolean;
}

export interface ReportState {
  validForm: boolean;
  edited: boolean;
  isLoading: boolean;
  error: boolean;
  step: number;
  reportData: ReportData;
  bookingInfo: BookingInfo;
}

export interface ReportData {
  id: number;
  booking_uuid: string;
  firewood: number;
  chopped_up_wood: number;
  smoke_detector_is_working: boolean;
  gas_is_full: boolean;
  gas_burner_primus: number;
  axe: number;
  hammer: number;
  saw: number;
  saw_blade: number;
  saw_bench: number;
  spade: number;
  kerosene_lamp: number;
  detergent: number;
  dishware: number;
  cookware: number;
  cabin_book: number;
  candle_holders: number;
  fire_blanket: number;
  fire_extinguisher: number;
  other_faults: string;
  boat_status: number;
  canoe_status: number;
  life_jackets_status: number;
  feedback: string;
}

export interface AvalancheState {
  warningLevels: Array<Warning>;
  disableBooking: boolean;
  warningData: Array<WarningResult>;
  koierToCheck: Array<string>;
}

export interface Warning {
  day: string;
  level: number;
}

export interface WarningResult {
  dangerLevel: number;
  validTo: string;
}

export interface CreateBookingInfo {
  koie: string;
  from_date: string;
  to_date: string;
  guests: Array<Guest>;
  guests_member: number;
  guests_not_member: number;
}

export interface BookingInfo {
  id: number;
  uuid: string;
  user: number;
  koie: string;
  price: number;
  booking_transaction_id: string;
  from_date: string;
  to_date: string;
  guests_member: number;
  guests_not_member: number;
}

export interface AuthState {
  isAdmin: boolean;
  tokens: {
    access: string | null;
    refresh: string | null;
  };
}

export interface Guest {
  name: string;
  number: string;
  email: string;
  isMember: boolean;
}

export interface AdminBookingsState {
  cabinsWithBookings: AdminBookingDictionary;
  startDate: string;
}

export interface AdminReportsState {
  chosenCabin: string;
  expanded: boolean;
  cabins: AdminReportKoieDictionary;
}
