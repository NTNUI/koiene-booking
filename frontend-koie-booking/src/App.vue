<template>
  <v-app :style="{ background: $vuetify.theme.themes['dark'].background }">
    <NavBar />

    <v-content>
      <router-view />
      <notifications position="top center" group="memberships_system" :ignore-duplicates="true" :max="1" />
    </v-content>
  </v-app>
</template>

<script lang="ts">
import Vue from 'vue';
import NavBar from './components/navBar/NavBar.vue';

export default Vue.extend({
  name: 'App',
  components: {
    NavBar,
  },
});
</script>

<style lang="scss">
@import 'styles/variables.scss';
</style>
