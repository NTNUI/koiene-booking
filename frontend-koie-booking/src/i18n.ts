import Vue from 'vue';
import VueI18n, { LocaleMessages } from 'vue-i18n';
import enJson from './locales/en.json';

Vue.use(VueI18n);

function loadLocaleMessages(): LocaleMessages {
  // Testing currently only supports manually importing of json-files. English is the only language that gets loaded now
  let locales: any;
  const messages: LocaleMessages = {};
  if (process.env.NODE_ENV === 'test') {
    messages['en'] = enJson;
  } else {
    locales = require.context('./locales', true, /[A-Za-z0-9-_,\s]+\.json$/i);
    locales.keys().forEach((key: any) => {
      const matched = key.match(/([A-Za-z0-9-_]+)\./i);
      if (matched && matched.length > 1) {
        const locale = matched[1];
        messages[locale] = locales(key);
      }
    });
  }
  return messages;
}

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || 'en',
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || 'en',
  messages: loadLocaleMessages()
});
