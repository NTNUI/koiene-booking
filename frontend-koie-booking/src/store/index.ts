import Vue from 'vue';
import Vuex, { StoreOptions } from 'vuex';
import axios from 'axios';
import VueAxios from 'vue-axios';
import { RootState } from './types';
import { koie } from './modules/users/koie/index';
import { booking } from './modules/users/booking/index';
import { avalanche } from './modules/users/avalanche/index';
import { report } from './modules/users/report/index';
import { auth } from './modules/users/auth/index';
import { adminBookings } from '@/store/modules/admin/bookings';

Vue.use(Vuex, VueAxios, axios);

const store: StoreOptions<RootState> = {
  modules: {
    adminBookings,
    koie,
    booking,
    avalanche,
    report,
    auth
  }
};

export default new Vuex.Store<RootState>(store);

export { store as storeConfig };
