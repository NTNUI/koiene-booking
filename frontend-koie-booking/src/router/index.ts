import Vue from 'vue';
import VueRouter from 'vue-router';
import store from '@/store';

Vue.use(VueRouter);

const routes = [
  {
    path: '/',
    name: 'AllKoier',
    component: () => import(/* webpackChunkName: "about" */ '../views/AllKoier.vue'),
    meta: { requiresAuth: false, requiresAdmin: false }
  },
  {
    path: '/koie/:id',
    name: 'koie',
    component: () => import('../views/Koie.vue'),
    props: true,
    meta: { requiresAuth: false, requiresAdmin: false }
  },
  {
    path: '/booking/:id',
    name: 'booking',
    component: () => import('../views/Booking.vue'),
    meta: { requiresAuth: true, requiresAdmin: false }
  },
  {
    path: '/login',
    name: 'Login',
    component: () => import('../views/Login.vue'),
    meta: { requiresAuth: false, requiresAdmin: false }
  },
  {
    path: '/admin',
    name: 'Admin',
    component: () => import('../views/Admin.vue'),
    meta: { requiresAuth: true, requiresAdmin: true }
  }
];

const router = new VueRouter({
  routes
});

export function beforeEach(to: any, from: any, next: any) {
  const isLoggedIn = store.getters['auth/isLoggedIn'];
  const isAdmin = store.getters['auth/isAdmin'];
  const requiresAuth = to.matched.some((record: any) => record.meta.requiresAuth);
  const requiresAdmin = to.matched.some((record: any) => record.meta.requiresAdmin);

  if (requiresAdmin && !isAdmin) next({ name: 'Login', query: { redirect: to.path } });
  else if (requiresAuth && !isLoggedIn) next({ name: 'Login', query: { redirect: to.path } });
  else if (to.name === 'Login' && isLoggedIn) next('/');
  else next();
}

router.beforeEach((to, from, next) => beforeEach(to, from, next));

export default router;
export { routes };
