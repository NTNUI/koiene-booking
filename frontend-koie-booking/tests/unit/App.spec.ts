import Vue from 'vue';
import Vuetify from 'vuetify';
import VueRouter from 'vue-router';
import Vuex from 'vuex';
import Notifications from 'vue-notification';
import { storeConfig } from '@/store';
import { cloneDeep } from 'lodash';
import i18n from '@/i18n';
import scssVars from '@/styles/variables.scss';
import mockAxios from 'jest-mock-axios';
import { routes } from '@/router';

Vue.use(Vuetify);

// Utilities
import { mount, createLocalVue } from '@vue/test-utils';

// Components or views
import App from '../../src/App.vue';

describe('App.vue', () => {
  const router = new VueRouter({ routes, mode: 'abstract' });
  let wrapper: any;
  let localVue: any;
  let vuetify: any;
  let store: any;

  beforeEach(() => {
    localVue = createLocalVue();
    vuetify = new Vuetify();
    localVue.use(VueRouter);
    localVue.use(Vuetify);
    localVue.use(Vuex);
    localVue.use(Notifications);
    // Hard resets the store between tests
    store = new Vuex.Store(cloneDeep(storeConfig));
    localVue.prototype.$scssVars = scssVars;
  });

  afterEach(() => {
    mockAxios.reset();
  });

  it('Matches snapshot', async () => {
    wrapper = mount(App, {
      localVue,
      router,
      vuetify,
      i18n,
      store
    });

    expect(wrapper).toMatchSnapshot();
  });
});
