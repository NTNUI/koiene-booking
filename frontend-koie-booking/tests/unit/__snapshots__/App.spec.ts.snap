// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue Matches snapshot 1`] = `
<div data-app="true" class="v-application v-application--is-ltr theme--light" id="app">
  <div class="v-application--wrap">
    <header class="v-sheet v-sheet--tile theme--dark v-toolbar v-app-bar v-app-bar--fixed backgroundLight" style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px;">
      <div class="v-toolbar__content" style="height: 64px;">
        <div><button target="_blank" type="button" class="v-btn v-btn--flat v-btn--text theme--dark v-size--default"><span class="v-btn__content"><h1 class="headline">NTNUI Koiene</h1></span></button></div>
        <div class="spacer"></div>
        <!----> <button type="button" class="mr-4 v-btn v-btn--contained theme--dark v-size--default"><span class="v-btn__content">
  LOG IN
</span></button>
      </div>
    </header>
    <main class="v-content" style="padding: 64px 0px 0px 0px;">
      <div class="v-content__wrap">
        <!---->
        <div class="vue-notification-group" style="width: 300px; top: 0px;"><span name="vn-fade"></span></div>
      </div>
    </main>
  </div>
</div>
`;
